src/main/java/com/epam/wl/executor/Executor.java

```diff
@@ -1,32 +1,29 @@
 package com.epam.wl.executor;
 
-import java.sql.Connection;
+import lombok.RequiredArgsConstructor;
+
+import javax.sql.DataSource;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
 
+@RequiredArgsConstructor
 public class Executor {
-    private final Connection connection;
-
-    public Executor(Connection connection) {
-        this.connection = connection;
-    }
+    private final DataSource dataSource;
 
     public void executeUpdate(final String update) throws SQLException {
-        final Statement stmt = connection.createStatement();
-        stmt.execute(update);
-        stmt.close();
+        try (Statement stmt = dataSource.getConnection().createStatement()) {
+            stmt.execute(update);
```

Comment:
I think  we need stmt.close(); here

Reply:
No, try-with-resources close resources automatically